pkg/git/git.go

```diff
@@ -0,0 +1,68 @@
+package git
+
+import (
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+)
+
+type IGitUtils interface {
```

Comment:
I recently learned that interfaces [should be defined on the consumer side](https://github.com/golang/go/wiki/CodeReviewComments#interfaces).

Reply:
I understand. I guess this is old behavior as it is handled like this in Java. :)

Reply:
done